packages/gen-ai-hub/src/client/openai/openai-response.ts

```diff
@@ -0,0 +1,47 @@
+import { HttpResponse } from "@sap-cloud-sdk/http-client";
+import { OpenAiChatCompletionOutput } from "./openai-types.js";
+
+/**
+ * Open AI chat completion response.
+ */
+export class OpenAiChatCompletionResponse {
+  public data: OpenAiChatCompletionOutput
+  constructor(private rawResponse: HttpResponse,) {
+    this.data = rawResponse.data;
+  }
+
+  /**
+   * Usage of tokens in the response.
+   */
+  get usage() {
+    return this.data.usage;
+  }
+  /**
+   * Reason for stopping the model.
+   */
+  get finish_reason() {
+    return this.data.choices[0].finish_reason;
+  }
+
+  /**
+* Parses the Open AI response and returns the content.
+* @param response - The  Open AI response.
+* @param choiceIndex - The index of the choice to parse.
+* @returns The message content.
+*/
+  getContent(choiceIndex: number = 0): string | null {
```

Comment:
[q] When would I get `null`? Maybe add this to the docs. It seems that we filter out `undefined`, so theoretically this can only happen when the message content itself really is there, but `null`. (see next comment)

Reply:
[req] I think we discussed that if there is a finish reason other than `success` or whatever the specific success message is, we would throw an error with the finish reason. Potentially this would also filter out the `null` messages (although, I thought those messages are usually empty string?)